test/test_torch.py

```diff
@@ -9742,6 +9742,14 @@ def _checked_swap(self, t1, t2):
         self.assertEqual(t1_moved, new_t2_moved)
         self.assertEqual(t2_moved, new_t1_moved)
 
+        # tests that PyObject slots on TensorImpl are correctly swapped by
+        # checking that when the function applied on a swapped tensor is
+        # returns doesn't change the TensorImpl, the returned value (which is
+        # given by returning the reference to the PyObject in the TensorImpl's
+        # PyObjectSlot) is still correct
+        self.assertEqual(id(t1.fill_(0.5)), id(t1))
```

Comment:
@albanD this is tested here! I'm not sure how to create a Tensor from Python where TensorImpl's `use_count > 1` or `weak_use_count > 1` to test that the checks are exercised though, do you know